openedx_rocketchat/api/v0/views.py

```diff
@@ -0,0 +1,233 @@
+
+
+class RocketChatSubscriptionsId(APIView):
+
+    authentication_classes = (
+        SessionAuthentication,
+    )
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def get(self, request):
+        """
+        Retrieve the rooms' id for every user subscription.
+
+        **Example Requests**:
+
+            GET /rocketchat/api/v0/subscriptions-id?unread=true&authToken=asdjfsdhfgasdas&&userId=86435f4asd56f4a
+
+            * unread : True to retrieve only subscription where the user has unread mentions.userId
+            * authToken : Rocketchat authorization token.
+            * userId : The unique rocketchat identifier for the user.
+
+        **Response Values**:
+
+            * subscriptions_id : A list with the room's id for every user subscription.
+
+        """
+        try:
+            unread = request.GET.get("unread")
+            auth_token = request.GET.get("authToken")
+            user_id = request.GET.get("userId")
+            serializer = RocketChatSubscriptionsIdSerializer(
+                data={
+                    "subscriptions_id": list(get_subscriptions_rids(auth_token, user_id, unread))
+                }
+            )
+            if serializer.is_valid():
+
+                return Response(serializer.data, status=status.HTTP_200_OK)
+
+        except Exception as error:
+            LOG.error("Rocketchat Subscriptions couldn't be got due to %s", error)
+            pass
+
+        return Response(serializer.data, status=status.HTTP_400_BAD_REQUEST)
+
+
+class RocketChatChangeRole(APIView):
+
+    authentication_classes = (
+        SessionAuthentication,
+    )
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request):
+        """
+        This methods allows to change the rocketchat role for a specific user
+
+        **Example Requests**:
+
+            POST /rocketchat/api/v0/change-role
+
+            * data = {
+                "username": "edx",
+                "role": "admin",
+            }
+
+        **Response Values**:
+
+            * It doesn't return data
+
+        """
+        if not request.user.is_staff:
+            return Response(status=status.HTTP_401_UNAUTHORIZED)
+
+        serializer = RocketChatChangeRoleSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response("Data is not valid", status=status.HTTP_400_BAD_REQUEST)
+
+        username = serializer.data["username"]
+        role = serializer.data["role"]
+
+        rocket_chat_settings = get_rocket_chat_settings()
+
+        api_rocket_chat = initialize_api_rocket_chat(rocket_chat_settings)
+
+        if api_rocket_chat:
+
+            user_info = api_rocket_chat.users_info(username=username)
+
+            try:
+                user_info = user_info.json()
+                if not user_info.get('success', False):
+                    return Response(user_info.get("error"), status=status.HTTP_400_BAD_REQUEST)
```

Comment:
A staff user can use this api to change the role for other user, that means if the staff user tries to change the role for a non-existent user, a message with that information should be shown because the site exist but the parameters on the request are not valid.

Reply:
Got it, but the response needs to be consistent returning a json format response, an then in the front you can parsed as needed. @andrey-canon